backends/pkg/reconciler/eventmesh/handler_test.go

```diff
@@ -0,0 +1,274 @@
+package eventmesh
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"knative.dev/pkg/apis"
+
+	"go.uber.org/zap"
+
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+
+	eventingv1 "knative.dev/eventing/pkg/apis/eventing/v1"
+	eventingv1beta2 "knative.dev/eventing/pkg/apis/eventing/v1beta2"
+
+	reconcilertestingv1 "knative.dev/eventing/pkg/reconciler/testing/v1"
+	reconcilertestingv1beta2 "knative.dev/eventing/pkg/reconciler/testing/v1beta2"
+
+	testingv1 "knative.dev/eventing/pkg/reconciler/testing/v1"
+	testingv1beta2 "knative.dev/eventing/pkg/reconciler/testing/v1beta2"
+)
+
+func TestBuildEventMesh(t *testing.T) {
+	tests := []struct {
+		name       string
+		brokers    []*eventingv1.Broker
+		eventTypes []*eventingv1beta2.EventType
+		want       EventMesh
+		error      bool
+	}{
+		{
+			name: "With 1 broker and 1 type",
+			brokers: []*eventingv1.Broker{
+				testingv1.NewBroker("test-broker", "test-ns",
+					// following fields are not used in any logic and simply returned
+					WithBrokerUID("test-broker-uid"),
+					WithBrokerLabels(map[string]string{"test-broker-label": "foo"}),
+					WithBrokerAnnotations(map[string]string{"test-broker-annotation": "foo"}),
+				),
+			},
+			eventTypes: []*eventingv1beta2.EventType{
+				testingv1beta2.NewEventType("test-eventtype", "test-ns",
+					testingv1beta2.WithEventTypeType("test-eventtype-type"),
+					testingv1beta2.WithEventTypeReference(brokerReference("test-broker", "test-ns")),
+					// following fields are not used in any logic and simply returned
+					testingv1beta2.WithEventTypeDescription("test-eventtype-description"),
+					WithEventTypeUID("test-eventtype-uid"),
+					WithEventTypeSchema(&apis.URL{
+						Scheme: "http",
+						Host:   "test-eventtype-schema",
+					}),
+					WithEventTypeSchemaData("test-eventtype-schema-data"),
+					testingv1beta2.WithEventTypeLabels(map[string]string{"test-eventtype-label": "foo"}),
+					WithEventTypeAnnotations(map[string]string{"test-eventtype-annotation": "foo"}),
+				),
+			},
+			want: EventMesh{
+				Brokers: []*Broker{
+					{
+						Name:               "test-broker",
+						Namespace:          "test-ns",
+						UID:                "test-broker-uid",
+						Labels:             map[string]string{"test-broker-label": "foo"},
+						Annotations:        map[string]string{"test-broker-annotation": "foo"},
```

Comment:
I'm wondering why we need to return labels and annotations here, is there a tagging concept in Backstage ?

Reply:
I've added a TODO on the plugin side to revisit that. Currently labels+annotations are not very useful, but it is possible to show pass these to Backstage.

Once I have some more progress (means 0.0.1 release of the plugin+backend), I will get a review from Backstage experts. I don't want to ask review prematurely.